dashboard/src/components/api/json-rpc/che-json-rpc-api.factory.ts

```diff
@@ -52,12 +52,16 @@ export class CheJsonRpcApi {
       return this.jsonRpcApiConnection.get(entrypoint);
     } else {
       const websocketClient = new WebsocketClient(this.$websocket, this.$q);
-      const cheJsonRpcMasterApi: CheJsonRpcMasterApi = new CheJsonRpcMasterApi(websocketClient, entrypoint, this.$log, this.$timeout, this.$interval, this.$q);
+      const cheJsonRpcMasterApi: CheJsonRpcMasterApi = new CheJsonRpcMasterApi(websocketClient, entrypoint, this.$log, this.$timeout, this.$interval, this.$q, this);
```

Comment:
I am sure it is better to use a callback **onConnectFail** or an anonymous function in the constructor.

But all this changes about recreation a new object if reconnection fails after 100 attempts(maxReconnectionAttempts = 100) with 30s deley after each (reconnectionDelay= 30000).  And it will call the same API ``` this.cheJsonRpcApi.connect(entrypoint) ``` and lost subscribers.

It is hard for me to image benefits after these changes, need more information.  If we have some problems, maybe it is better to change values for **maxReconnectionAttempts** and/or **reconnectionDelay** and that is it.

@monaka WDYT?